config/nrfconnect/nrfconnect-app.cmake

```diff
@@ -88,6 +85,30 @@ find_file(CHIP_PROJECT_CONFIG
     NO_DEFAULT_PATH
 )
 
+find_file(OPENTHREAD_OVERLAY_CONFIG
+    OpenThreadOverlayConfig.h
+    PATHS ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/main/include ${CHIP_ROOT}/config/nrfconnect
+    NO_DEFAULT_PATH
+)
+
+# ==================================================
+# Configure Zephyr
+# ==================================================
+
+# Add dependency on nRF crypto library.
+# This is temporary solution and should be removed after fixing it in nRF Connect SDK.
+zephyr_include_directories("${ZEPHYR_BASE}/../nrfxlib/crypto/nrf_cc310_platform/include")
+
+# Apply OpenThread config overrides by defining OPENTHREAD_CORE_CONFIG_PLATFORM_CHECK_FILE.
+# OPENTHREAD_CORE_CONFIG_PLATFORM_CHECK_FILE, which might seem more accurate, cannot be used
+# since it's already used by Zephyr.
+if (OPENTHREAD_OVERLAY_CONFIG)
+    get_filename_component(OPENTHREAD_OVERLAY_DIR ${OPENTHREAD_OVERLAY_CONFIG} DIRECTORY)
+    get_filename_component(OPENTHREAD_OVERLAY_FILE ${OPENTHREAD_OVERLAY_CONFIG} NAME)
+    target_include_directories(ot-config INTERFACE ${OPENTHREAD_OVERLAY_DIR})
+    target_compile_definitions(ot-config INTERFACE OPENTHREAD_CORE_CONFIG_PLATFORM_CHECK_FILE="${OPENTHREAD_OVERLAY_FILE}")
```

Comment:
Oh, this looks like a BIG hack :smile:. I see that this will work, but the OpenThread Core Config Check file is used for different purpose. And the need of doing #undefs is a little bit ugly.

OpenThread stack expects application/project to define the custom configuration by defining `OPENTHREAD_PROJECT_CORE_CONFIG_FILE` define. For Zephyr it is defined and hardcoded here: 
https://github.com/zephyrproject-rtos/openthread/blob/zephyr/zephyr/CMakeLists.txt#L198

Maybe the above approach has good reasoning, but i believe the `OT_CONFIG` settings should be used instead:
https://github.com/zephyrproject-rtos/openthread/blob/d216ec32cdb78ec873689234f710b3c9bb096bcd/CMakeLists.txt#L132

Like here: 
https://github.com/zephyrproject-rtos/openthread/blob/d216ec32cdb78ec873689234f710b3c9bb096bcd/examples/platforms/cc2538/CMakeLists.txt#L31

I would do one of two:
1. Overwrite OPENTHREAD_PROJECT_CORE_CONFIG_FILE define, and provide a full OpenThread config suitable for us (this will involve copying certain parts from openthread-core-zephyr-config.h. This may require some changes in openthread repo - I'm not sure if we can simply remove this -DOPENTHREAD_... flag. If we can, maybe we can use `OT_CONFIG` in CHIP.

2. Modify `openthread-core-zephyr-config.h` by allowing it to include custom application config e.g. on top of the file:
```
#ifndef OPENTHREAD_APP_PROJECT_CORE_CONFIG_FILE
#include OPENTHREAD_APP_PROJECT_CORE_CONFIG_FILE
#endif
```

And define `OPENTHREAD_APP_PROJECT_CORE_CONFIG_FILE` in our build process.

@Damian-Nordic  @kkasperczyk-no thoughts?

Reply:
You're right about `OPENTHREAD_PROJECT_CORE_CONFIG_FILE` and `OT_CONFIG` but the problem is that Zephyr initializes `OPENTHREAD_PROJECT_CORE_CONFIG_FILE` in a quite dumb way using `add_definitions()` which pollutes configuration of all OT subdirectories. If I try to set `OT_CONFIG` I get warnings about macro redefinition, hence I decided to go with `OPENTHREAD_CORE_CONFIG_PLATFORM_CHECK_FILE`.

Solution 1 would obviously be great, but I would need to clean many targets to be able to overwrite it without the redefinition problem which may not be easy to do if we want to avoid "hacks", but I'll spend more time on that.

Solution 2 would also be nice, but requires changes in Zephyr and I think it would be nice to provide a solution which would work with official NCS 1.4, right?

Reply:
Ok. I found less hacky way to do it :)